File: smartmedika.admin/src/app/model/regionModel.ts
import { IEntityBase }          from './IEntityBase';
import { CountryModel }         from './countryModel';
import { UTCTimeBaseModel }     from './utcTimeBaseModel';
import { RegencyModel }         from './regencyModel';

export class RegionModel implements IEntityBase{
    
    id: number;
    name: string;
    countryId: number;
    utcId: number;
    

   
    country: CountryModel;
    utcTime: UTCTimeBaseModel; 
    regencyList: RegencyModel[];   

    constructor(){
        this.id = 0;
        this.name = null;
        this.countryId = 0;
        this.utcId - 0;
        this.country = null
        this.utcId = 0;
        this.country = null
        this.utcTime = null
        this.regencyList = null
    }     

}

File: server/medicore.layer/MediCore.Service/Model/RegionModel.cs
using dna.core.data.Infrastructure;
using dna.core.service.Models.Abstract;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace MediCore.Service.Model
{
    public class RegionModel : IModelBase
    {
        public int Id { get; set; }
        [Required(ErrorMessage = "Required")]
        [StringLength(100, ErrorMessage = "StringLength")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Required")]
        public int CountryId { get; set; }
        [Required(ErrorMessage = "Required")]
        public int UTCId { get; set; }

        [Required]
        public Status Status { get; set; }

        public virtual CountryModel Country { get; set; }

       
        public virtual UTCTimeBaseModel UTCTime { get; set; }

        public virtual ICollection<RegencyModel> Regencys { get; set; }
    }
}
